pkg/remote/aws/route_table_association_supplier.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/cloudskiff/driftctl/pkg/parallel"
+	"github.com/cloudskiff/driftctl/pkg/remote/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	awsdeserializer "github.com/cloudskiff/driftctl/pkg/resource/aws/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+	"github.com/hashicorp/terraform/flatmap"
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type RouteTableAssociationSupplier struct {
+	reader       terraform.ResourceReader
+	deserializer deserializer.CTYDeserializer
+	client       ec2iface.EC2API
+	runner       *terraform.ParallelResourceReader
+}
+
+func NewRouteTableAssociationSupplier(runner *parallel.ParallelRunner, client ec2iface.EC2API) *RouteTableAssociationSupplier {
+	return &RouteTableAssociationSupplier{
+		terraform.Provider(terraform.AWS),
+		awsdeserializer.NewRouteTableAssociationDeserializer(),
+		client,
+		terraform.NewParallelResourceReader(runner),
+	}
+}
+
+func (s RouteTableAssociationSupplier) Resources() ([]resource.Resource, error) {
+
+	tables, err := listRouteTables(s.client)
+	if err != nil {
+		logrus.Error(err)
```

Comment:
I saw this a lot on my pr. It's not a really good practice in go to log the error and return it.
You might want to wrap it to add this to the stacktrace but there is already a caller that log the same error.

Reply:
Good catch, I fixed this one.
How do you handle context in errors ? If I want to return an error but with local context to be visible on caller log ?

![image](https://user-images.githubusercontent.com/6154987/105177043-5f425680-5b26-11eb-9fd3-bdcd1af29d02.png)

Reply:
You can wrap the error https://godoc.org/github.com/pkg/errors#Wrap
But this only allow to add to the stacktrace and error message.

Reply:
A custom error type may be a solution too ?

Reply:
Is this something that we could formalize in terms of kind of error ? As @eliecharra said, if we find a 403 error we could "Wrap" the err in a `forbidden` error ?

Reply:
> A custom error type may be a solution too ?

yep we could start making our own error type and add a context map for our logger for example. But still wrapping is cool that way we could print the error stack when debugging

Reply:
> Is this something that we could formalize in terms of kind of error ? As @eliecharra said, if we find a 403 error we could "Wrap" the err in a `forbidden` error ?

yep it's a common pattern to have type or singleton to identify error source.
But in what I did 403 are completely hidden to the outside.